### PR TITLE
PMM-6552 Before register, check if config file is writable

### PR DESCRIPTION
**Pull request by Dasio**
https://jira.percona.com/browse/PMM-6552

Also when node already exists, inform user about `--force` option

FB: https://github.com/Percona-Lab/pmm-submodules/pull/1748

### CODE REVIEW

File: config/config_windows.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	"os"
+)
+
+// IsWritable tries to open and write comment to file.
+func IsWritable(path string) error {
+	file, err := os.OpenFile(path, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0640)
+	if err != nil {
+		return err
+	}
+	_, err = file.Write([]byte("# Write check\n"))
```
Comment (Dasio) [inline config/config_windows.go:33]: maybe this write is not necessary?

File: config/config_windows.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
+// pmm-agent
```
Comment (BupycHuk) [inline config/config_windows.go:1]: Do we support running pmm client in windows OS. As far as I know we don't.

File: commands/setup.go [hunk 3]
```diff
@@ -62,6 +63,10 @@ func Setup() {
 	}
 
 	if !cfg.Setup.SkipRegistration {
+		if err := config.IsWritable(configFilepath); err != nil {
```
Comment (BupycHuk) [inline commands/setup.go:66]: we can extract it from this `if`. command may fail even with `--skip-registation` flag and it's better to have the same message in both cases.

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: config/config_windows.go (hunk 2 above)
Comment (Dasio) [reply config/config_windows.go:1]: So let's delete both files (_unix,_windows), move to config.go and use just unix package?

Comment (BupycHuk) [reply config/config_windows.go:1]: yeah

**Review by BupycHuk — APPROVED** [review]

File: commands/setup.go [hunk 4]
```diff
@@ -61,6 +62,11 @@ func Setup() {
 		os.Exit(1)
 	}
 
+	if err := config.IsWritable(configFilepath); err != nil {
+		fmt.Printf("Config file %s is not writable: %s.\n", configFilepath, err)
```
Comment (artemgavrilov) [inline commands/setup.go:66]:
```suggestion
		fmt.Printf("Config file %s is not writable: %v.\n", configFilepath, err)
```

**Review by artemgavrilov — APPROVED** [review]

File: commands/setup.go (hunk 4 above)
Comment (artemgavrilov) [reply commands/setup.go:66]: Technically both correct, but golang team uses `%v` in their examples:
https://blog.golang.org/error-handling-and-go

Also in our codebase `%v` used for errors printing.

**Pull request merged**